### PR TITLE
Fixes for AlgorithmProperties and small bug fixes.

### PR DESCRIPTION
**Pull request by TomFischer**
Extracted commits of PR #897

### CODE REVIEW

**Comment by bilke** [comment]
:+1:

File: MeshLib/MeshGenerators/VtkMeshConverter.cpp [hunk 1]
```diff
@@ -265,8 +265,8 @@ MeshLib::Mesh* VtkMeshConverter::constructMesh(const double* pixVal,
 	if (value_vec->empty())
 		properties.removePropertyVector("Colour");
 
-	boost::optional< MeshLib::PropertyVector<unsigned>& > materials =
-		properties.createNewPropertyVector<unsigned>("MaterialIDs", MeshLib::MeshItemType::Cell, 1);
+	boost::optional< MeshLib::PropertyVector<int>& > materials =
+		properties.createNewPropertyVector<int>("MaterialIDs", MeshLib::MeshItemType::Cell, 1);
 	materials->insert(materials->end(), elements.size(), 0);
```
Comment (endJunction) [inline MeshLib/MeshGenerators/VtkMeshConverter.cpp]: There is a little more code at this place in the original PR...

**Comment by endJunction** [comment]
One missing change aside, :fast_forward:

File: MeshLib/MeshGenerators/VtkMeshConverter.cpp (hunk 1 above)
Comment (TomFischer) [inline MeshLib/MeshGenerators/VtkMeshConverter.cpp]: That is true. I will integrate further changes in one of the next PR.

**Pull request merged**